components/connectionmanager/src/lib.rs

```diff
@@ -227,12 +227,14 @@ impl ConnectionManager {
         }
 
         if missing_connections > 0 && !self.dns_seeders.is_empty() {
-            let cmgr = self.clone();
-            // DNS lookup is a blocking i/o operation, so we spawn it as a blocking task
-            let _ = tokio::task::spawn_blocking(move || {
-                cmgr.dns_seed(missing_connections); //TODO: Consider putting a number higher than `missing_connections`.
-            })
-            .await;
+            if missing_connections > self.outbound_target / 2 {
+                // If we are missing more than half of our target, query all in parallel.
+                // This will always be the case on new node start-up and is the most resilient strategy in such a case.
+                self.dns_seed_many(self.dns_seeders.len()).await;
+            } else {
+                // Try to obtain at least twice the number of missing connections
+                self.dns_seed_sequential(2 * missing_connections).await;
```

Comment:
naming is not clear, both variants adds many connections. 
my suggestion is `dns_seed_many` -> `dns_seed_in_parallel`

Reply:
They are also different in the way they decide to break (min addresses or num seeders), I think parallel vs. seq is misleading. Any other better suggestion?

Reply:
`_by_seeders[_count]` and `_by_connections[_count]`

Reply:
Done